routes/progress.js

```diff
@@ -0,0 +1,98 @@
+require("dotenv").config();
+const express = require("express");
+const router = express.Router();
+let progressDB = require("../progressDatabase.js");
+const jwt = require("jsonwebtoken");
+
+router.get("/", authenticateToken, (req, res) => {
+  let userId = 1; //This should not be hard coded. Should get userId from JWT in future
+
+  let select = "SELECT language,category,progress FROM progress WHERE userId=?";
+  let params = [userId];
+  const rowData = [];
+  progressDB.all(select, params, (error, rows) => {
+    if (error) {
+      console.log(error);
+      res.status(500).send();
+    }
+    rows.forEach((row) => {
+      rowData.push(row);
+    });
+    res.json(rowData).send();
+  });
+});
+
+router.post("/update", async (req, res) => {
+  const language = req.body.language;
+  const category = req.body.category;
+  const userId = 1; //Hard coded for now.
+  const progress = 30;
+
+  //Check if it exists
+  let select =
+    "SELECT progress FROM progress WHERE userId=? AND language=? AND category=?";
+  let params = [userId, language, category];
+  const rowData = [];
+  progressDB.all(select, params, (error, rows) => {
+    if (error) {
+      console.log(error);
+      res.status(500).send();
+    }
+    rows.forEach((row) => {
+      rowData.push(row);
+    });
+    if (rowData.length === 0) {
+      addNewProgress(res, userId, language, category);
+    } else {
+      currentProgress = rowData[0].progress;
+      newProgress = currentProgress + 10;
+      if (newProgress <= 100) {
+        updateProgress(res, newProgress, userId, language, category);
+      } else {
+        res.status(200).send();
+      }
+    }
+  });
+});
+
+const addNewProgress = (res, userId, language, category) => {
```

Comment:
Perhaps we want to check if the language/category progress added is implemented or not? Or perhaps that is irrelevant at this point?